students/alex_whitty/session05/mailroom3.py

```diff
@@ -0,0 +1,165 @@
+import sys  # imports go at the top of the file
+
+
+
+# Data Structure
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Elon Musk": [2263.23, 3300.87, 15432.0],
+            }
+
+dict_comprehension = \
+     {d: donors for d, donors in donor_db.items() if donors}
+print(dict_comprehension)
+
+# Input/Output
+prompt = "\n".join(("Mail Distribution Center!",
+          "Please choose from below options:",
+          "1 - Send a Thank You to a single donor",
+          "2 - Create a Report",
+          "3 - Send letters to all donors",
+          "4 - Exit",
+          ">>> "))
+
+# Features
+
+dict_comp = {d for d in donor_db.keys()}
+print(dict_comp)
+
+
+
+
+def view_donor_names():
+
+    for donor in donor_db.keys():
+        print(donor)
+
+
+def write_a_letter(donor, amount):
+    letter = f"Dear {donor},\n\nThank you for your kind donation of ${str(amount)}\n\n" \
+        f"It will be put to very good use.\n\n" \
+        f"Sincerely,\n" \
+        f"The Fundraising Committee"
+    return letter
+
+def write_a_letter_all(donor):
+    letter = f"Dear {donor},\n\nThank you for your kind donation. \n\n" \
+        f"It will be put to very good use.\n\n" \
+        f"Sincerely,\n" \
+        f"The Fundraising Committee"
+    return letter
+
+''' Switch with the above functions "write_a_letter and write_a_letter_all" '''
+
+switch_func_dict = {
+    0: write_a_letter,
+    1: write_a_letter_all,
+}
```

Comment:
This should be up at the top or bottom but not in among functions where it can get lost.